sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md

```diff
@@ -225,35 +253,57 @@ Now in `Azure.Messaging.ServiceBus`, we simplfify this by giving session variant
-ServiceBusSessionProcessor processor = client.CreateSessionProcessor(queueName);
-
-// create a processor to receive events from the given set of sessions
+```C# Snippet:ServiceBusConfigureSessionProcessor
+// get the options to use for configuring the processor
 var options = new ServiceBusSessionProcessorOptions
 {
-    SessionIds = ["my-session", "your-session"],
-};
-ServiceBusSessionProcessor processor = client.CreateSessionProcessor(queueName, options);
+    // By default after the message handler returns, the processor will complete the message
+    // If I want more fine-grained control over settlement, I can set this to false.
+    AutoCompleteMessages = false,
 
-// create a processor to receive events from the 3 next available sessions
-var options = new ServiceBusSessionProcessorOptions
-{
-    MaxConcurrentSessions = 3
+    // I can also allow for processing multiple sessions
+    MaxConcurrentSessions = 5,
+
+    // By default, there will be a single concurrent call per session. I can
+    // increase that here to enable parallel processing within each session.
+    MaxConcurrentCallsPerSession = 2,
+
+    // Processing can be optionally limited to a subset of session Ids.
+    SessionIds = { "my-session", "your-session" },
 };
-ServiceBusSessionProcessor processor = client.CreateSessionProcessor(queueName);
+
+// create a session processor that we can use to process the messages
+ServiceBusSessionProcessor processor = client.CreateSessionProcessor(queueName, options);
 ```
 
 The below code snippet shows you the session variation of the receiver. Please note that creating a session receiver is an async operation because the library will need to get a lock on the session by connecting to the service first.
 
-```cs
-// create a receiver to receive events from the next available session
+Create a receiver that will receive from the next available session:
+```C# Snippet:ServiceBusReceiveNextSession
 ServiceBusSessionReceiver receiver = await client.AcceptNextSessionAsync(queueName);
 
-// create a receiver to receive events from the given session
-ServiceBusSessionReceiver receiver = await client.AcceptSessionAsync(queueName, "my-session");
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveMessageAsync();
+Console.WriteLine(receivedMessage.SessionId);
 ```
 
+Create a receiver that will receive from a specific session:
+```C# Snippet:ServiceBusReceiveFromSpecificSession
+// create a receiver specifying a particular session
+ServiceBusSessionReceiver receiver = await client.AcceptSessionAsync(queueName, "Session2");
+
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveMessageAsync();
+Console.WriteLine(receivedMessage.SessionId);
+```
+
+## Known Gaps from Previous Library
```

Comment:
In the individual bullet points, lets use `the previous library, Microsoft.Azure.ServiceBus` rather than just `the previous library`

Reply:
How about just including the name for each bullet point, but saying both previous and the name in the summary sentence above?

Reply:
Eh I will just include the whole thing for each one.